src/posix/platform/infra_if.cpp

```diff
@@ -60,6 +61,47 @@
 #include "lib/platform/exit_code.h"
 #include "posix/platform/infra_if.hpp"
 
+otError otPlatInfraIfGetHostAddress(const char *aHostName, otIp6Address *aAddress)
+{
+    otError          error = OT_ERROR_NOT_FOUND;
+    struct addrinfo  hints;
+    struct addrinfo *res = nullptr;
+    int              status;
+
+    memset(&hints, 0, sizeof hints);
+    hints.ai_family   = AF_INET6;
+    hints.ai_socktype = SOCK_STREAM;
+
+    if ((status = getaddrinfo(aHostName, NULL, &hints, &res)) != 0)
```

Comment:
I think `getaddrinfo` might block for a very long time, but it's inproper for the OT main loop to halt, which may cause packet loss or serious timing issues.
Maybe we should try `getaddrinfo_a`. 

Your thoughts? @bukepo

Reply:
Good point. Let me take a look at using `getaddrinfo_a`.

Reply:
Updated with `getaddrinfo_a` to request address lookup and handle the lookup result asynchronously. When the lookup completes, `InfraNetif::HandleHostAddressResponse` is invoked to process the result and notify routing manager the address. PTAL. Thanks!